homeassistant/components/homekit_controller/climate.py

```diff
@@ -170,8 +182,56 @@ def get_characteristic_types(self) -> list[str]:
             CharacteristicsTypes.TEMPERATURE_COOLING_THRESHOLD,
             CharacteristicsTypes.TEMPERATURE_HEATING_THRESHOLD,
             CharacteristicsTypes.SWING_MODE,
+            CharacteristicsTypes.ROTATION_SPEED,
         ]
 
+    @property
+    def fan_modes(self) -> list[str] | None:
+        """Return the available fan modes."""
+        if self.service.has(CharacteristicsTypes.ROTATION_SPEED):
+            return [FAN_OFF, FAN_LOW, FAN_MEDIUM, FAN_HIGH]
+        return None
+
+    @property
+    def fan_mode(self) -> str | None:
+        """Return the current fan mode."""
+        speed_range = (round(self.service[CharacteristicsTypes.ROTATION_SPEED].minValue or 0) + 1,
+                       round(self.service[CharacteristicsTypes.ROTATION_SPEED].maxValue or 100))
+        speed_percentage = ranged_value_to_percentage(
+            speed_range, self.service.value(
+                CharacteristicsTypes.ROTATION_SPEED)
+        )
+        fan_mode = FAN_OFF
+
+        # homekit value 0 33 66 100
+        if speed_percentage > ROTATION_SPEED_MEDIUM:
+            fan_mode = FAN_HIGH
+        elif speed_percentage > ROTATION_SPEED_LOW:
+            fan_mode = FAN_MEDIUM
+        elif speed_percentage > 0:
+            fan_mode = FAN_LOW
+
+        return fan_mode
+
+    async def async_set_fan_mode(self, fan_mode: str) -> None:
+        """Set new target fan mode."""
+        rotation = 0
+        if fan_mode == FAN_LOW:
+            rotation = ROTATION_SPEED_LOW
+        elif fan_mode == FAN_MEDIUM:
+            rotation = ROTATION_SPEED_MEDIUM
+        elif fan_mode == FAN_HIGH:
+            rotation = ROTATION_SPEED_HIGH
+
+        speed_range = (round(self.service[CharacteristicsTypes.ROTATION_SPEED].minValue or 0) + 1,
```

Comment:
This could be a function since it's duplicated code with the above

Reply:
ok fixed